sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_properties.asciidoc

```diff
@@ -320,6 +344,23 @@ class handler {
                                range<dimensions> workGroupSize,
                                PropertyList properties,
                                const WorkgroupFunctionType &kernelFunc);
+
+  // Available only if `!has_compile_time_kernel_effect<PropertyT>` for all
+  // properties `PropertyT` in `PropertyList`.
+  template <typename PropertyList>
+  void single_task(PropertyList properties, const kernel& kernelObject);
+
+  // Available only if `!has_compile_time_kernel_effect<PropertyT>` for all
+  // properties `PropertyT` in `PropertyList`.
+  template <int Dimensions, typename PropertyList>
+  void parallel_for(range<Dimensions> numWorkItems, PropertyList properties,
+                    const kernel& kernelObject);
+
+  // Available only if `!has_compile_time_kernel_effect<PropertyT>` for all
+  // properties `PropertyT` in `PropertyList`.
+  template <int Dimensions, typename PropertyList>
+  void parallel_for(nd_range<Dimensions> ndRange, PropertyList properties,
+                    const kernel& kernelObject);
```

Comment:
I also wonder if we should invest effort into extending these forms of `single_task` and `parallel_for`.  The recently implemented [sycl_ext_oneapi_enqueue_functions](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc) has a cleaner separation between launch properties vs. kernel properties.

I think our longer term strategy may be to drop the forms of `single_task` and `parallel_for` in sycl_ext_oneapi_kernel_properties and use the ones in [sycl_ext_oneapi_enqueue_functions](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc) instead.

@Pennycook what were your thoughts on these two extension regarding ways of specifying kernel properties?

Reply:
> I think our longer term strategy may be to drop the forms of single_task and parallel_for in sycl_ext_oneapi_kernel_properties and use the ones in [sycl_ext_oneapi_enqueue_functions](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc) instead.

I agree with this.

sycl_ext_oneapi_kernel_properties was essentially our first attempt at defining a kernel launch interface that accepted properties.  The sycl_ext_oneapi_enqueue_functions design addresses a lot of feedback we received from users and other implementers, and it's much more aligned with where we expect future versions of SYCL to go.

My preference would be that we move to sycl_ext_oneapi_enqueue_functions as soon as possible, and deprecate these property overloads.  If there's a short-term need to expose these overloads with minimal effort, I'm not opposed, but we should figure out an implementation plan for the new extension.

Reply:
> If there's a short-term need to expose these overloads with minimal effort, I'm not opposed, but we should figure out an implementation plan for the new extension.

I think [sycl_ext_oneapi_enqueue_functions](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc) is already implemented.

@steffenlarsen I'm not sure who is asking for this change.  Can we ask them to use sycl_ext_oneapi_enqueue_functions instead of making the changes in this PR?